### DISCUSSION TITLE
dm+d codelist download

### DISCUSSION THREAD

**Discussion by rc16** [opensafely/documentation#1227] [Codelists]
Hi,

Sorry if this is a silly question! 

When I download the dm+d codelists it seems to cut off the code, even if I format the cells to number in excel. This means when I re-upload the csv's the codes are not quite correct (I don't think). 

For example I downloaded some of the opioid codelists e.g. https://www.opencodelists.org/codelist/opensafely/opioid-containing-medicines-oral-excluding-drugs-for-substance-misuse-dmd/2fb91091/#full-list, as needed to regroup this and some of the other opioid codelists to different levels of weak and strong opioids e.g. https://www.opencodelists.org/codelist/opensafely/opioids-strong-for-msk-conditions/34c95431/ But when I cross-reference the downloaded codelist with the original online, or the new uploaded one the codes are a bit different.. Is there any around this?

Ruth

**Comment by Jongmassey** [comment]
Hi Ruth,
Unfortunately there's no real good way to coerce Excel into not mangling dm+d codes, so my recommendation is to not open dm+d codelist (and probably SNOMED-CT) files in Excel at all. I'll update the documentation accordingly to reiterate this point.

**Comment by Jongmassey** [comment]
If you absolutely must use Excel, then you may find that rather  than opening the csv file, you can open a new Excel file and then import the file which lets you specify the file types on import. Using Text rather than Number for the dmd id fields may leave them intact (it's the conversion to Excel's native numeric type that causes most of the problems)

**Comment by rc16** [comment]
Importing as text has worked, I do need it to update my codelists. 

Thanks.
